akka23-conductr-client-lib/src/main/scala/com/typesafe/conductr/clientlib/akka/BundlesConnector.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright © 2014-2016 Typesafe, Inc. All rights reserved.
+ * No information contained herein may be reproduced or transmitted in any form
+ * or by any means without the express written permission of Typesafe, Inc.
+ */
+
+package com.typesafe.conductr.clientlib.akka
+
+import akka.actor.{ ActorRef, FSM, Props }
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.pattern._
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Flow
+import com.typesafe.conductr.clientlib.akka.models.{ EventStreamFailure, EventStreamResult, EventStreamSuccess }
+import com.typesafe.conductr.clientlib.scala.models.Bundle
+import de.heikoseeberger.akkasse.ServerSentEvent
+
+import scala.concurrent.duration._
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success, Try }
+
+/** Connects to the /bundles endpoint of ConductR and triggers the given [[com.typesafe.conductr.clientlib.akka.BundlesConnector.Callback]] when updates are available. */
+object BundlesConnector {
+
+  type GetBundleEvents = () => Future[EventStreamResult]
+  type GetBundlesInfo = () => Future[Seq[Bundle]]
+  /**
+   * Represents the callback to be invoked when event burst is completed.
+   * Returns [[CallbackResult.Stop]] to indicate completion, i.e. no further action is required, [[BundlesConnector]] will be shut down.
+   * Returns [[CallbackResult.Continue]] to continue with listening for events.
+   */
+  type Callback = Seq[Bundle] => CallbackResult
```

Comment:
I think I'd like to avoid these type aliases as they require further cognitive load when browsing the source. Explicitly declaring `() => Future[EventStreamResult]` as a param type is more obvious than `GetBundleEvents`.